src/interface/sparse_pack.hpp

```diff
@@ -437,8 +340,71 @@ class SparsePack : public SparsePackBase {
     const int vidx = GetLowerBound(b, t) + t.idx;
     return pack_(dir, b, vidx)(k, j, i);
   }
+
+  template <class... VTs>
+  auto GetPtrs(const int b, const TE el, int k, int j, int i, VTs... vts) {
+    return std::make_tuple({&(*this)(b, el, vts, k, j, i)...});
+  }
 };
 
+inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<std::string> &vars,
+                               const std::vector<bool> &use_regex,
+                               const std::vector<MetadataFlag> &flags = {},
+                               const std::set<PDOpt> &options = {}) {
+  PARTHENON_REQUIRE(vars.size() == use_regex.size(),
+                    "Vargroup names and use_regex need to be the same size.");
+  auto selector = [&](int vidx, const VarID &id, const Metadata &md) {
+    if (flags.size() > 0) {
+      for (const auto &flag : flags) {
+        if (!md.IsSet(flag)) return false;
+      }
+    }
+
+    if (use_regex[vidx]) {
+      if (std::regex_match(std::string(id.label()), std::regex(vars[vidx]))) return true;
+    } else {
+      if (vars[vidx] == id.label()) return true;
+      if (vars[vidx] == id.base_name && id.sparse_id != InvalidSparseID) return true;
+    }
+    return false;
+  };
+
+  impl::PackDescriptor base_desc(psd, vars, selector, options);
+  return typename SparsePack<>::Descriptor(base_desc);
+}
+
+template <class... Ts>
+inline auto MakePackDescriptor(StateDescriptor *psd,
+                               const std::vector<MetadataFlag> &flags = {},
+                               const std::set<PDOpt> &options = {}) {
+  static_assert(sizeof...(Ts) > 0, "Must have at least one variable type for type pack");
+
+  std::vector<std::string> vars{Ts::name()...};
+  std::vector<bool> use_regex{Ts::regex()...};
+
+  return typename SparsePack<Ts...>::Descriptor(static_cast<impl::PackDescriptor>(
+      MakePackDescriptor(psd, vars, use_regex, flags, options)));
+}
+
+inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<std::string> &vars,
+                               const std::vector<MetadataFlag> &flags = {},
+                               const std::set<PDOpt> &options = {}) {
+  return MakePackDescriptor(psd, vars, std::vector<bool>(vars.size(), false), flags,
+                            options);
+}
+
+inline auto MakePackDescriptor(
```

Comment:
Can we also get a UID overload for the pack descriptor?

Reply:
```c++
inline auto MakePackDescriptor(StateDescriptor *psd, const std::vector<Uid_t> &vars,
                               const std::vector<MetadataFlag> &flags = {},
                               const std::set<PDOpt> &options = {}) {
  
  auto selector = [&](int vidx, const VarID &id, const Metadata &md) {
    if (flags.size() > 0) {
      for (const auto &flag : flags) {
        if (!md.IsSet(flag)) return false;
      }
    }
      if (vars[vidx] == Variable::GetUniqueID(id.label())) return true;
    }
    return false;
  };
  std::string var_str; 
  for (auto v : vars) {var_str.push_back(std::to_string(v));} 
  impl::PackDescriptor base_desc(psd, var_str, selector, options);
  return typename SparsePack<>::Descriptor(base_desc);
}

Reply:
I think the optional flag vector can be dropped too. It can be assumed they've already been filtered.

Reply:
I think we should maybe think about splitting the `MakePackDescriptor` functions out into their own file so that they are easy to find. Maybe we can do this in another PR and add a few different options for making pack descriptors.

Reply:
:+1: that sounds good